### PR TITLE
Reports table updates

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
Changes reports model to make lat and long un-nullable, and changes resource to send the fields through the validation method upon report creation. Also changes validation method to .strip the field value only if it is a string. Also changes the migration to make the attrs un-nullable but it is not necessary to drop / re-migrate for functionality to take effect. Sad path tests for missing lat and long updated, sadpath tests for empty fields removed because you can't really have an empty float.

- [x] Updated Files: api/database/models.py, api/resources/reports.py, tests/endpoints/reports/test_create_report.py

- [ ] Routes Added:

- [x] Sad Path Tested

- [x] Happy Path Tested

- [x] All Tests are passing

- [ ] Updated README.md with added information

- [x] Collaborators: @PhilipDeFraties

### CODE REVIEW

File: api/resources/reports.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 
 def _validate_field(data, field, proceed, errors, missing_okay=False):
     if field in data:
-        data[field] = data[field].strip()
+        if type(data[field]) is str:
+            data[field] = data[field].strip()
```
Comment (Todd-Estes) [inline api/resources/reports.py:13]: 👍

**Review by Todd-Estes — APPROVED** [review]

File: tests/endpoints/reports/test_create_report.py [hunk 2]
```diff
@@ -103,6 +114,52 @@ def test_happypath_blank_longitude(self):
         data = json.loads(response.data.decode('utf-8'))
         assert_payload_field_type_value(self, data, 'success', bool, True)
 
+    # def test_happypath_missing_image(self):
```
Comment (Oxalisviolacea) [inline tests/endpoints/reports/test_create_report.py:117]: Is this test failing?

**Review by Oxalisviolacea — APPROVED** [review]

**Review by araspaas — APPROVED** [review]
